prompt_doctor/debug_tool.py

```diff
@@ -1,65 +1,103 @@
-from flask import Flask, render_template, request, jsonify
-from .prompt_manager import PromptManager
-from .llm_call import call_llm_api
+import json
+import logging
 import threading
 import webbrowser
-import json
+from pathlib import Path
+
+from flask import Flask, render_template, request, redirect
+from jinja2 import Template
+from openai import OpenAI
+from openai.types.chat.chat_completion import ChatCompletion
+from werkzeug.serving import make_server
+
+logger = logging.getLogger(__name__)
 
 
 class DebugTool:
-    def __init__(self, prompts_dir="prompts"):
-        self.app = Flask(__name__)
-        self.prompt_manager = PromptManager(prompts_dir)
-        self.current_prompt_id = None
-        self.current_initial_prompt = None
-        self.current_context = None
-
-        @self.app.route("/", methods=["GET", "POST"])
+
+    def __init__(self, oai_client: OpenAI, prompts_dir="prompts"):
+        self.llm_client = oai_client
+        self.prompts_dir = Path(prompts_dir)
+
+    def _call_llm(self, prompt: str, **oai_kwargs) -> ChatCompletion:
+        # TODO: extend template to support multiple messages/roles
```

Comment:
I was thinking of was passing previous_messages and the prompt_id and then create the list from that. But extending the template could work as well -- then if there's a list of messages they could be passed in via the context.

Reply:
This way is more flexible/general, for example: if you have a list of messages and want to create a new prompt with it, you can pass them in the context and interpolate in the template (as you mentioned); but if the API accepts a list of messages and always prepend to the prompt (as history), then is not possible to change the prompt to ignore the history (since prepending happens after).